### PR TITLE
AC-657 Added About Section

### PR DESCRIPTION
**Pull request by tbagz104**
## Description of what I changed
Since a basic info about the community and the goal behind the app was missing, I think an About section regarding the above mentioned factors is necessary for a socially useful productive app. I've added the About section page as well as social media, website redirects to the page.

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-657

## Screenshots
snap

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java [hunk 1]
```diff
@@ -206,6 +207,12 @@ public boolean onOptionsItemSelected(MenuItem item) {
                 Observable<List<Location>> observableList = new LocationDAO().getLocations();
                 observableList.subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).subscribe(getLocationList());
                 return true;
+
+            case R.id.actionAbout:
+                Intent about = new Intent(this, AboutActivity.class);
+                startActivity(about);
+
+
```
Comment (VibhorChinda) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java]: @tbagz104 i think you can remove this extra space.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    @Override
+    public void onClick(View v) {
+
+        switch (v.getId()) {
+
+            case R.id.facebookButton:
+                v.startAnimation(buttonClick);
+                startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse("https://www.facebook.com/OpenMRS/")));
```
Comment (VibhorChinda) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java]: @tbagz104 i guess you should make the links as a string resource and refer them from the strings.xml file.

and please remove the extra spacing present in this file

File: openmrs-client/src/main/res/layout-land/activity_about.xml [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+		app:layout_constraintBottom_toBottomOf="parent"
+		app:layout_constraintEnd_toEndOf="parent"
+		app:layout_constraintHorizontal_bias="0.683"
+		app:layout_constraintStart_toStartOf="parent"
+		app:layout_constraintTop_toBottomOf="@+id/aboutText"
+		app:layout_constraintVertical_bias="0.751"
+		app:srcCompat="@drawable/ic_youtube"
+		tools:ignore="MissingConstraints,VectorDrawableCompat" />
+
+
```
Comment (VibhorChinda) [inline openmrs-client/src/main/res/layout-land/activity_about.xml]: remove extra space

File: openmrs-client/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -360,4 +360,10 @@
     <string name="intro_settings_desc">Try out our dark mode feature.\nExciting? Let\'s get started.</string>
     <string name="get_started_with_our_user_guide">Get started with our user guide</string>
 
+	<!-- About Activity -->
+	<string name="about">About</string>
+	<string name="aboutText">The OpenMRS community is a worldwide network of volunteers from many different backgrounds including technology, health care, and international development. \n\nTogether, we\'re working to build the world\'s largest and most flexible technology platform to support delivery of health care in some of the\nmost challenging environments on the planet.</string>
```
Comment (VibhorChinda) [inline openmrs-client/src/main/res/values/strings.xml]: @tbagz104 i guess should follow snake case not camel case here.
Like
about_text

File: openmrs-client/src/main/res/values/styles.xml [hunk 5]
```diff
@@ -113,5 +113,24 @@
...
+
+    <style name="MyShapeAppearance">
+        <item name="cornerFamily">rounded</item>
+        <item name="cornerSizeTopRight">32dp</item>
+        <item name="cornerSizeBottomRight">32dp</item>
+        <item name="cornerSizeTopLeft">32dp</item>
+        <item name="cornerSizeBottomLeft">32dp</item>
+
+    </style>
+
```
Comment (VibhorChinda) [inline openmrs-client/src/main/res/values/styles.xml]: please remove extra spacing from this file too.

**Comment by tbagz104** [comment]
@f4ww4z @ribhavsharma @dkayiwa Could any one of you please review my PR? Thanks in advance! :)

**Comment by tbagz104** [comment]
@VibhorChinda thanks for pointing out. I've made the changes. @rishabh-997 @f4ww4z can you guys please review my PR?

File: openmrs-client/src/main/res/layout/fragment_patient_details.xml [hunk 6]
```diff
@@ -19,230 +19,241 @@
 	android:layout_width="match_parent"
 	android:layout_height="match_parent">
 
-	<androidx.constraintlayout.widget.ConstraintLayout
```
Comment (rishabh-997) [inline openmrs-client/src/main/res/layout/fragment_patient_details.xml]: Pull the recent changes, these are changes from previous commits...

File: openmrs-client/src/main/res/layout/activity_provider_dashboard.xml [hunk 7]
```diff
@@ -94,7 +94,7 @@
             app:tabIndicatorColor="@color/white"
             app:tabMode="fixed"
             app:tabGravity="fill"
-            app:tabSelectedTextColor="@color/black"
```
Comment (rishabh-997) [inline openmrs-client/src/main/res/layout/activity_provider_dashboard.xml]: revert the changes, or pull from master again

File: openmrs-client/src/main/res/layout/activity_patient_dashboard.xml [hunk 8]
```diff
@@ -54,7 +54,7 @@
             android:layout_height="wrap_content"
             app:tabIndicatorColor="@color/white"
             app:tabMode="scrollable"
-            app:tabSelectedTextColor="@color/black"
+            app:tabSelectedTextColor="@color/white"
```
Comment (rishabh-997) [inline openmrs-client/src/main/res/layout/activity_patient_dashboard.xml]: same here

File: openmrs-client/src/main/res/layout-land/activity_about.xml [hunk 9]
```diff
@@ -0,0 +1,122 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (rishabh-997) [inline openmrs-client/src/main/res/layout-land/activity_about.xml]: There us already activity_about.xml ??? why saving separately in layout_land folder ???

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+import android.view.animation.AlphaAnimation;
+import android.widget.Button;
+import android.widget.ImageButton;
+import androidx.appcompat.app.AppCompatActivity;
+import android.os.Bundle;
+import androidx.appcompat.app.AppCompatDelegate;
+import org.openmrs.mobile.R;
+import org.openmrs.mobile.utilities.ThemeUtils;
+
+public class AboutActivity extends AppCompatActivity implements View.OnClickListener {
```
Comment (rishabh-997) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java]: Extend it to ACBaseActivity

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+                startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.youtube_url))));
+                break;
+
+            default:
+                break;
+        }
+
+    }
+
+    public void setupTheme(){
```
Comment (rishabh-997) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java]: Already available in ACBase Activity

File: openmrs-client/src/main/res/layout-land/activity_about.xml (hunk 9 above)
Comment (tbagz104) [reply openmrs-client/src/main/res/layout-land/activity_about.xml]: I would like to create a landscape variation for the same, since the default one doesn't look like good UI.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java (hunk 11 above)
Comment (tbagz104) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java]: Oh I'll make the changes. Thanks :)

File: openmrs-client/src/main/res/layout-land/activity_about.xml (hunk 9 above)
Comment (rishabh-997) [reply openmrs-client/src/main/res/layout-land/activity_about.xml]: okay. check the other comments

**Comment by tbagz104** [comment]
@rishabh-997 I've made the changes. Hope everything is alright now :)

File: openmrs-client/src/main/res/layout/fragment_patient_details.xml [hunk 12]
```diff
@@ -19,230 +19,241 @@
...
+                android:layout_marginStart="@dimen/margin_short"
+                android:layout_marginLeft="@dimen/margin_short"
+                android:layout_marginTop="30dp"
+                android:textColor="?android:attr/textColorPrimary"
+                android:textSize="@dimen/text_size_large"
+                android:textStyle="bold"
+                app:layout_constraintStart_toEndOf="@+id/patientPhotoCardView"
+                app:layout_constraintTop_toTopOf="parent" />
+
+            <ImageView
```
Comment (rishabh-997) [inline openmrs-client/src/main/res/layout/fragment_patient_details.xml]: why is this here ??? this has been removed in recent commits

Comment (tbagz104) [reply openmrs-client/src/main/res/layout/fragment_patient_details.xml]: My Bad I've made the changes. Please take a look :)

**Comment by tbagz104** [comment]
@rishabh-997 Though the Travis CI  build shows failed, when i check out the tests, the build is successful. Is there any way the build can be checked again? Looks like i don't have access to build again. 

Screenshot from 2020-02-16 17-30-10

File: openmrs-client/src/main/res/layout/fragment_patient_details.xml (hunk 12 above)
Comment (rishabh-997) [reply openmrs-client/src/main/res/layout/fragment_patient_details.xml]: You need to pull the latest changes from master. Try 'git pull --rebase upstream master' and fix any merge conflicts if any. Currently you are including changes from previous merges in your PR

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+                startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.youtube_url))));
+                break;
+
+            default:
+                break;
+        }
+
+    }
+
+    private void initialize(){
```
Comment (CaptainDaVinci) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java]: There should be a space after the closing brace.

File: openmrs-client/src/main/res/values/color.xml [hunk 14]
```diff
@@ -43,4 +43,6 @@
 
     <!-- neutral colors that look pleasing on both white and black background -->
     <color name="neutral_green">#249627</color>
+    <color name="blueBtn">#3F51B5</color>
```
Comment (CaptainDaVinci) [inline openmrs-client/src/main/res/values/color.xml]: To keep the names consistent with other colors already present in the file, I think the name "blueBtn" should be changed to something like "dark_blue" or any other name which better describes the color.

File: openmrs-client/src/main/res/values/styles.xml [hunk 15]
```diff
@@ -113,5 +113,21 @@
         <item name="android:textColor">#8affffff</item>
     </style>
 
+    <style name="AppTheme.Button" parent="Widget.AppCompat.Button.Borderless.Colored">
+        <item name="colorButtonNormal">@color/blueBtn</item>
+        <item name="android:textColor">@color/white</item>
+    </style>
+
+    <style name="MyButton" parent="Widget.MaterialComponents.Button.OutlinedButton">
```
Comment (CaptainDaVinci) [inline openmrs-client/src/main/res/values/styles.xml]: Can a better style name be used instead of "MyButton", which could provide an idea about the use of that style?

File: openmrs-client/src/main/res/values/styles.xml [hunk 16]
```diff
@@ -113,5 +113,21 @@
...
+        <item name="shapeAppearanceOverlay">@style/MyShapeAppearance</item>
+    </style>
+
+    <style name="MyShapeAppearance">
```
Comment (CaptainDaVinci) [inline openmrs-client/src/main/res/values/styles.xml]: Here as well.

File: openmrs-client/src/main/res/values-hi/strings.xml [hunk 17]
```diff
@@ -359,4 +359,10 @@
 	<string name="intro_settings_desc">हमारी डार्क मोड सुविधा देखें।\n आएँ शुरू करें।</string>
 	<string name="get_started_with_our_user_guide">हमारे उपयोगकर्ता गाइड के साथ आरंभ करें</string>
 
+	<!-- About Activity -->
+	<string name="about">के बारे में</string>
+	<string name="about_text">OpenMRS समुदाय प्रौद्योगिकी, स्वास्थ्य देखभाल और अंतर्राष्ट्रीय विकास सहित कई विभिन्न पृष्ठभूमि के स्वयंसेवकों का एक विश्वव्यापी नेटवर्क है। \ n \ n पूरी तरह से, हम ग्रह पर कुछ सबसे चुनौतीपूर्ण वातावरण में स्वास्थ्य देखभाल के वितरण का समर्थन करने के लिए दुनिया के सबसे बड़े और सबसे लचीले प्रौद्योगिकी मंच का निर्माण करने के लिए काम कर रहे हैं।</string>
+	<string name="website">वेबसाइट</string>
+
```
Comment (CaptainDaVinci) [inline openmrs-client/src/main/res/values-hi/strings.xml]: There are some blank lines here which can be removed.

**Comment by f4ww4z** [comment]
@tbagz104 Did you see @CaptainDaVinci 's review? Also please see the failing Travis CI build.

**Comment by tbagz104** [comment]
@f4ww4z I'm in the middle of exams, so I'm AFK at the moment. I have seen the comments and I will make the changes and push :)

**Comment by tbagz104** [comment]
@f4ww4z I've made the changes as requested by @CaptainDaVinci. Also I've resolved issues with the failing Travis-CI build. I hope everything is solved :) Could you please look into it?

**Comment by tbagz104** [comment]
@f4ww4z sir can you please look into this? Thanks in advance :)

File: openmrs-client/src/main/res/layout/activity_about.xml [hunk 18]
```diff
@@ -0,0 +1,121 @@
...
+		app:layout_constraintEnd_toEndOf="parent"
+		app:layout_constraintStart_toStartOf="parent"
+		app:layout_constraintTop_toBottomOf="@+id/aboutText"
+		app:layout_constraintVertical_bias="0.581"
+		app:shapeAppearance="@style/WebButtonShapeAppearance"
+		app:strokeColor="@color/dark_purple"
+		app:strokeWidth="6dp"
+		tools:ignore="MissingConstraints" />
+
+	<ImageButton
```
Comment (AryanGanotra07) [inline openmrs-client/src/main/res/layout/activity_about.xml]: In activity_about.xml , for all three ImageButtons following the common style, you should add a style in styles.xml if there is repeating style for any view.

Comment (tbagz104) [reply openmrs-client/src/main/res/layout/activity_about.xml]: @AryanGanotra07 Thanks for the review mate, I'll make the changes :)

File: openmrs-client/src/main/AndroidManifest.xml [hunk 19]
```diff
@@ -113,6 +113,9 @@
             android:launchMode="singleTop"
             android:parentActivityName=".activities.dashboard.DashboardActivity"
             android:theme="@style/AppThemeOrig" />
+        <activity
+            android:name=".activities.about.AboutActivity"
+            android:parentActivityName=".activities.about.AboutActivity"/>
```
Comment (rishabh-997) [inline openmrs-client/src/main/AndroidManifest.xml]: Add a label and link it the string to string resource file
Add a theme to it so it responds to light-dark mode

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+
+public class AboutActivity extends ACBaseActivity implements View.OnClickListener {
+
+    private AlphaAnimation buttonClick = new AlphaAnimation(1F, 0.8F);
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_about);
+        setupTheme();
```
Comment (rishabh-997) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java]: remove this line since ACBaseActivity takes care for this

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+
+            default:
+                break;
+        }
+
+    }
+
+    private void initialize() {
+
+        ImageButton facebookButton = (ImageButton) findViewById(R.id.facebookButton);
```
Comment (rishabh-997) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java]: TypeCasting is redundant.
Also use [View Binding](https://developer.android.com/topic/libraries/view-binding) now onwards

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java [hunk 22]
```diff
@@ -0,0 +1,83 @@
...
+        facebookButton.setOnClickListener(this);
+        ImageButton twitterButton = (ImageButton) findViewById(R.id.twitterButton);
+        twitterButton.setOnClickListener(this);
+        ImageButton youtubeButton = (ImageButton) findViewById(R.id.youtubeButton);
+        youtubeButton.setOnClickListener(this);
+        Button websiteButton = (Button) findViewById(R.id.websiteButton);
```
Comment (rishabh-997) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/about/AboutActivity.java]: see above comments, do this for all your views.

File: openmrs-client/src/main/res/values-hi/strings.xml [hunk 23]
```diff
@@ -359,4 +359,8 @@
 	<string name="intro_settings_desc">हमारी डार्क मोड सुविधा देखें।\n आएँ शुरू करें।</string>
 	<string name="get_started_with_our_user_guide">हमारे उपयोगकर्ता गाइड के साथ आरंभ करें</string>
 
+	<!-- About Activity -->
+	<string name="about">के बारे में</string>
```
Comment (rishabh-997) [inline openmrs-client/src/main/res/values-hi/strings.xml]: change this translation, sounds absurd for a title

File: openmrs-client/src/main/res/values/strings.xml [hunk 24]
```diff
@@ -360,4 +360,12 @@
     <string name="intro_settings_desc">Try out our dark mode feature.\nExciting? Let\'s get started.</string>
     <string name="get_started_with_our_user_guide">Get started with our user guide</string>
 
+	<!-- About Activity -->
+	<string name="about">About</string>
+	<string name="about_text">The OpenMRS community is a worldwide network of volunteers from many different backgrounds including technology, health care, and international development. \n\nTogether, we\'re working to build the world\'s largest and most flexible technology platform to support delivery of health care in some of the\nmost challenging environments on the planet.</string>
+	<string name="website">VISIT US</string>
```
Comment (rishabh-997) [inline openmrs-client/src/main/res/values/strings.xml]: the key-value pairs do not seem similar, change either as per functionality it performs.

File: openmrs-client/src/main/res/layout/fragment_patient_details.xml [hunk 25]
```diff
@@ -14,235 +14,235 @@
...
-			android:layout_height="1dp"
-			android:layout_marginTop="50dp"
-			android:background="@color/transparent"
-			android:visibility="visible"
-			app:layout_constraintStart_toStartOf="parent"
-			app:layout_constraintTop_toBottomOf="@+id/addressDetailsCountry" />
-
-	</androidx.constraintlayout.widget.ConstraintLayout>
-</ScrollView>
+    xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (rishabh-997) [inline openmrs-client/src/main/res/layout/fragment_patient_details.xml]: use `git pull --rebase upstream master` to get the latest code. There is no change in this `fragment_patient_details.xml`.

**Comment by f4ww4z** [comment]
@tbagz104 did you address the above comments?

**Comment by tbagz104** [comment]
@f4ww4z Yes I've seen the comments and I will make the changes as requested :)

**Comment by f4ww4z** [comment]
@tbagz104 Also kindly fix the merge conflicts as seen here.

**Comment by rishabh-997** [comment]
any updates on this ???

**Comment by tbagz104** [comment]
> any updates on this ???

I'm working on the changes and will resolve the conflicts :)

**Comment by f4ww4z** [comment]
> I'm working on the changes and will resolve the conflicts :)

@tbagz104 Any progress on this?

**Comment by f4ww4z** [comment]
Closing due to inactivity.
